File: lab4/libsvm-3.22/matlab/svm2.m
function svm_2()
% Load training features and labels
[y, x] = libsvmread('ex8Data/ex8a.txt');
gamma = 100;

% Libsvm options
% -s 0 : classification
% -t 2 : RBF kernel
% -g : gamma in the RBF kernel
model = svmtrain(y, x, sprintf('-s 0 -t 2 -g %g', gamma));

% Display training accuracy
[predicted_label, accuracy, decision_values] = svmpredict(y, x, model);

% Plot training data and decision boundary
plotboundary(y, x, model);
title(sprintf('\\gamma = %g', gamma), 'FontSize', 14);

% ex8b.txt ============================================================
% Load training features and labels
[y, x] = libsvmread('ex8Data/ex8b.txt');
gamma = 100;

% Libsvm options
% -s 0 : classification
% -t 2 : RBF kernel
% -g : gamma in the RBF kernel
model = svmtrain(y, x, sprintf('-s 0 -t 2 -g %g', gamma));

% Display training accuracy
[predicted_label, accuracy, decision_values] = svmpredict(y, x, model);

% Plot training data and decision boundary
plotboundary(y, x, model);
title(sprintf('\\gamma = %g', gamma), 'FontSize', 14);

File: lab5/svm2.py
# -*- coding: utf-8 -*-
# @File     : svm2.py
# @Author   : Yanjie Ke
# @Date     : 2019/1/13 12:36
# @License  : Copyright(C), USTC

from svmutil import *

print('ex8a.txt')
y, x = svm_read_problem('libsvm-3.23/python/ex8Data/ex8a.txt')

split_num = int(len(y) * 0.66)
train = svm_problem(y[:split_num], x[:split_num], isKernel=True)

'''
gamma = 1 / (2 * σ^2)
'''
gamma = 650
model = svm_train(train, '-s 0 -t 2 -g %s' % gamma)
svm_predict(y[split_num:], x[split_num:], model)

print('ex8b.txt')
y, x = svm_read_problem('libsvm-3.23/python/ex8Data/ex8b.txt')

split_num = int(len(y) * 0.66)
train = svm_problem(y[:split_num], x[:split_num], isKernel=True)

'''
gamma = 1 / (2 * σ^2)
'''
gamma = 650
model = svm_train(train, '-s 0 -t 2 -g %s' % gamma)

svm_predict(y[split_num:], x[split_num:], model)


